src/core.c/ThreadPoolScheduler.pm6

```diff
@@ -151,7 +151,9 @@ my class ThreadPoolScheduler does Scheduler {
                             }
                             if $resume {
                                 nqp::push($!queue, {
-                                    nqp::continuationinvoke($continuation, nqp::null())
+                                    $l.protect: {
+                                        nqp::continuationinvoke($continuation, nqp::null())
+                                    }
```

Comment:
We may be able to make this more efficient.

1. I think we only need to use the lock for synchronization purposes if `nqp::isconcrete($continuation)` comes back false.
2. I wonder if we really need to protect the invoke operation, or if we just want to do a `$l.lock; $l.unlock;` beforehand.